LabExT/View/Controls/Plotting/PlotController.py

```diff
@@ -104,11 +108,149 @@ def __draw_axes_line_plot(self, axis_x_key: str, axis_y_key: str) -> None:
         plot.set_xlabel(axis_x_key)
         plot.set_ylabel(axis_y_key)
 
+    def __draw_axes_contour_plot(self) -> None:
+        """Creates a contour or contourf plot on the plotting frame."""
+        if len(self._plottable_data) < 2:
+            # for a contour plot we need at least 2 selected measurements
+            return
+
+        if "" in [
+            self._model.axis_x_key_name.get(),
+            self._model.axis_y_key_name.get(),
+            self._model.axis_z_key_name.get(),
+        ]:
+            # the variables are set to "" by the settings frame if something doesn't work
```

Comment:
can you be more specific? What's happening if "something doesnt work"?

Reply:
Added more information in 1a3cf748c9c7a0dcfa5e5cc136615e2406d647a8.